File: March 8, 2026.js
function isValidHSL(hsl) {

  if (!hsl.startsWith("hsl(") || !(hsl.endsWith(")") || hsl.endsWith(";"))) {
    return false;
  }

  const content = hsl.slice(hsl.indexOf("(") + 1, hsl.lastIndexOf(")"));
  const arr = content.split(",").map(x => x.trim());

  if (parseInt(arr[0]) < 0 || parseInt(arr[0]) > 360) {
    return false;
  }

  if (!arr[1].endsWith("%") || !arr[2].endsWith("%")) {
    return false;
  }

  const numberS = parseInt(arr[1].slice(0, -1));
  const numberL = parseInt(arr[2].slice(0, -1));

  if (numberS < 0 || numberS > 100) {
    return false;
  }

  if (numberL < 0 || numberL > 100) {
    return false;
  }

  return true;
}

File: March 8, 2026.py
def is_valid_hsl(hsl):

    if not hsl.startswith("hsl(") or not hsl.endswith((")", ";")):
        return False

    content = hsl[hsl.find("(")+1 : hsl.rfind(")")]
    arr = [x.strip() for x in content.split(",")]

    if int(arr[0]) < 0 or int(arr[0]) > 360:
        return False

    if not arr[1].endswith("%") or not arr[2].endswith("%"):
        return False
    
    number_s = int(arr[1][:-1])
    number_l = int(arr[2][:-1])

    if number_s < 0 or number_s > 100:
        return False

    if number_l < 0 or number_l > 100:
        return False

    return True